src/redux/missions/missions.js

```diff
@@ -0,0 +1,38 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+
+const getUrl = 'https://api.spacexdata.com/v3/missions';
+
+const FETCHED_MISSION = 'FETCHED_MISSION';
+const JOIN_OR_LEAVE_MISSION = 'JOIN_OR_LEAVE_MISSION';
+
+export const getMissions = createAsyncThunk(
+    FETCHED_MISSION,
+    async () => {
+      const missionArr = [];
+     await fetch(getUrl).then((missions) => missions.json())
+      .then(res => res.forEach((mission) => {
+        missionArr.push({
+            mission_id: mission.mission_id,
+            mission_name: mission.mission_name,
+            description: mission.description,
+            reserved: false
+        });
+      }));
+      return (missionArr);
+    },
+  );
```

Comment:
The getMissions thunk could be improved for readability. The use of async/await syntax mixed with .then syntax is not typical and could be confusing. You should consider using just async/await like this:

export const getMissions = createAsyncThunk(
    'missions/fetchMissions',  // String for the action type
    async () => {
        const response = await fetch(getUrl);
        const missionsData = await response.json();
        const missionArr = missionsData.map(mission => ({
            mission_id: mission.mission_id,
            mission_name: mission.mission_name,
            description: mission.description,
            reserved: false
        }));
        return missionArr;
    },
);